Mathlib/LinearAlgebra/UnitaryGroup.lean

```diff
@@ -176,13 +176,25 @@ def embeddingGL : unitaryGroup n α →* GeneralLinearGroup α (n → α) :=
 
 open scoped Kronecker in
 /-- The kronecker product of two unitary matrices is unitary. -/
-theorem kronecker_mem_unitaryGroup {m : Type*} [Fintype m] [DecidableEq m]
-    {U₁ : Matrix n n α} {U₂ : Matrix m m α}
-    (hU₁ : U₁ ∈ unitaryGroup n α) (hU₂ : U₂ ∈ unitaryGroup m α) :
-    U₁ ⊗ₖ U₂ ∈ unitaryGroup (n × m) α := by
-  simp_rw [mem_unitaryGroup_iff, star_eq_conjTranspose, conjTranspose_kronecker,
-    ← mul_kronecker_mul, ← star_eq_conjTranspose, mem_unitaryGroup_iff.mp hU₁,
-    mem_unitaryGroup_iff.mp hU₂, one_kronecker_one]
+theorem kronecker_mem_unitaryGroup {R m : Type*}
+    [Semiring R] [StarRing R] [Fintype m] [DecidableEq m]
+    {U₁ : Matrix n n R} {U₂ : Matrix m m R}
+    (hU₁ : U₁ ∈ unitary (Matrix n n R)) (hU₂ : U₂ ∈ unitary (Matrix m m R)) :
+    U₁ ⊗ₖ U₂ ∈ unitary (Matrix (n × m) (n × m) R) := by
+  simp_rw [unitary.mem_iff, star_eq_conjTranspose, conjTranspose_kronecker']
+  constructor <;> ext <;> simp [Matrix.mul_apply,
+    ← Matrix.star_apply, ← mul_assoc]
+  · simp_rw [mul_assoc _ (star U₁ _ _), ← Finset.univ_product_univ, Finset.sum_product]
+    rw [Finset.sum_comm]
+    simp_rw [← Finset.sum_mul, ← Finset.mul_sum, ← Matrix.mul_apply, hU₁.1,
+      Matrix.one_apply, mul_boole, ite_mul, zero_mul,
+      Finset.sum_ite_irrel, ← Matrix.mul_apply, hU₂.1,
+      Matrix.one_apply, Finset.sum_const_zero, ← ite_and, Prod.eq_iff_fst_eq_snd_eq]
+  · simp_rw [mul_assoc _ _ (star U₂ _ _), ← Finset.univ_product_univ, Finset.sum_product,
+      ← Finset.sum_mul, ← Finset.mul_sum, ← Matrix.mul_apply, hU₂.2,
+      Matrix.one_apply, mul_boole, ite_mul, zero_mul,
+      Finset.sum_ite_irrel, ← Matrix.mul_apply, hU₁.2,
+      Matrix.one_apply, Finset.sum_const_zero, ← ite_and, and_comm, Prod.eq_iff_fst_eq_snd_eq]
```

Comment:
This probably can now go above the `UnitaryGroup` namespace, since it doesn't actually mention `unitaryGroup`. The name should then be `kronecker_mem_unitary`.